### PR TITLE
Validate limits and requests if resource quota is enabled in project

### PR DESCRIPTION
**Pull request by loganhz**
## Proposed changes

Validate limits and requests if resource quota is enabled in project.

As the limits and requests are in the advanced option which is hidden by default, so it's better to expand the security section if the error happens. Otherwise, it's hard for users to find out where should they input the limits and requests for cpu/memory.

## Types of changes

What types of changes does your code introduce to Rancher?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Issues

https://github.com/rancher/rancher/issues/15508

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
_Tip: `yarn lint` or `npm run lint`_
- [ ] Any dependent issues or pr's have been linked
- [ ] If updating dependencies, `yarn.lock` file has been updated and committed

## Further comments

n/a

### CODE REVIEW

File: app/components/container/new-edit/component.js [hunk 1]
```diff
@@ -204,6 +205,16 @@ export default Component.extend(NewOrEdit, ChildHook, {
...
     let pr = get(this, 'primaryResource');
     let errors = pr.validationErrors() || [];
+    const lc = get(this, 'launchConfig');
+
+    const quotaErrors = lc.validateQuota() || [];
+
+    errors.pushObjects(quotaErrors);
+
+    if ( get(quotaErrors, 'length') > 0 ) {
+      set(this, 'advanced', true);
```
Comment (westlywright) [inline app/components/container/new-edit/component.js]: `setProperties(...)`

File: app/models/container.js [hunk 2]
```diff
@@ -38,6 +39,37 @@ var Container = Resource.extend(DisplayImage, {
...
+    const projectLimit =  get(this, 'scope.currentProject.resourceQuota.limit');
+
+    if ( !projectLimit ) {
+      return [];
+    }
+
+    const intl = get(this, 'intl');
+    const errors = [];
+
+    const limitsCpu = get(projectLimit, 'limitsCpu');
```
Comment (westlywright) [inline app/models/container.js]: You can utilize destructuring assignment here. Its not required but I figured you may not know about it yet and it makes the code a little more concise imo. 

```
const { limitsCpu, limitsMemory, requestCpu, requestMemory } = projectLimit;
```

Comment (westlywright) [reply app/models/container.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

File: app/models/workload.js [hunk 3]
```diff
@@ -345,6 +345,12 @@ Workload.reopenClass({
       })
     }
 
+    if ( data && data.containers ) {
```
Comment (westlywright) [inline app/models/workload.js]: Be consistent with your gets, utilize the framework.

File: app/components/container/new-edit/component.js [hunk 4]
```diff
@@ -204,6 +205,16 @@ export default Component.extend(NewOrEdit, ChildHook, {
   validate() {
     let pr = get(this, 'primaryResource');
     let errors = pr.validationErrors() || [];
+    const lc = get(this, 'launchConfig');
+
+    const quotaErrors = lc.validateQuota() || [];
```
Comment (westlywright) [inline app/components/container/new-edit/component.js]: validateQuota returns an empty array if `projectLimit` doesn't exist or an empty errors array. I think the extra defensive array here is unneeded.

File: app/components/container/new-edit/component.js [hunk 5]
```diff
@@ -204,6 +205,16 @@ export default Component.extend(NewOrEdit, ChildHook, {
   validate() {
     let pr = get(this, 'primaryResource');
     let errors = pr.validationErrors() || [];
+    const lc = get(this, 'launchConfig');
```
Comment (westlywright) [inline app/components/container/new-edit/component.js]: validateQuota returns an empty array if projectLimit doesn't exist or an empty errors array. I think the extra defensive array here is unneeded.

**Comment by loganhz** [comment]
@westlywright  Done. Thanks

**Pull request merged**